Plot images at their native resolution, without any resizing

alexdu opened the pull request:
Reason: when you are working with images, you usually want to see the result exactly as you have computed it, not resized or otherwise messed up with.

With this, whatever you want to display using imshow will appear in the notebook at the same resolution as the original image, without any resizing.

Image data is scaled, so the display range parameter works as expected. It's always saved as 8-bit, to keep things simple (didn't see any reason not to).

Limitations:

- huge images are not downsized, so they will slow down (and maybe lock up) the browser
- colormaps, axes, other annotations will not work, so you may want to make this optional
- I doubt SVG will work (not tested), but I'm not sure what to do about it

Test notebook [here](https://dl.dropboxusercontent.com/u/4124919/bleeding-edge/octave_kernel/octave-kernel-image-tests.html).

Another example notebook [here](https://dl.dropboxusercontent.com/u/4124919/bleeding-edge/stripefix/stripe-fix-h264.html) (I wrote this patch while doing that experiment => now you can see a possible use case for this feature)

That's it.

blink1073 commented:
Hi @alexdu,  there is no way to differentiate between an image produced by `imshow` and any other plotting command.  I think the correct thing to do is to only use the `resolution` parameter, and default to 150 (the same as Octave).

alexdu commented:
I used the "cdata" property to differentiate between them. Regular plots don't have it, so the "try" block will fail.

blink1073 commented:
Thanks!  I didn't pick up on your trick there.  What happens for you if we just use `print` and do not specify `resolution` at all?  We could only set the parameter if explicitly overridden.

alexdu commented:
I updated to current version, which removed my changes:
```sudo pip install octave_kernel --upgrade --force-reinstall```

Then I modified the "print" line to read:
```print(_handle, [_filename, '.%(fmt)s']);```

and I re-executed the test notebook. [Result](https://dl.dropboxusercontent.com/u/4124919/bleeding-edge/octave_kernel/octave-kernel-image-tests-no-res.html).

Note: after upgrading, but before changing the line, the result looked exactly the same.

BTW, when printing a figure that contains an image, octave also displays the axis (so the image has some border) and always scales the image to figure size. I want a way to disable the scaling completely.

blink1073 commented:
Okay, I'm sold on this as-is.  Thanks, @alexdu!

blink1073 commented:
Released as 0.16.0.
